### PR TITLE
9-Automatically-generating-REST-endpoint's-key-fields

### PR DESCRIPTION
**Pull request by 1AhmedYasser**

### CODE REVIEW

File: GUI/src/components/MainNavigation/index.tsx
Comment (trevorling) [inline GUI/src/components/MainNavigation/index.tsx:1]: Let's use single quotes. 
It's hard to do the review if there are a lot of changes that are related to the double quotes change.

File: GUI/src/constants/menuIcons.tsx
Comment (trevorling) [inline GUI/src/constants/menuIcons.tsx:1]: Let's use single quotes.
It's hard to do the review if there are a lot of changes that are related to the double quotes change.

File: GUI/src/pages/NewServicePage.tsx [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+  const [openApiSpec, setOpenApiSpec] = useState("");
+  const [jsonSchema, setJsonSchema] = useState<any>();
+  const [selectedSchema, setSelectedSchema] = useState<any>();
+  const [endpoints, setEndpoints] = useState<EndpointType[]>([]);
+  const [selectedEndpoint, setSelectedEndpoint] = useState<string>("");
+
+  const { t } = useTranslation();
+
+  useEffect(() => {
+    if (jsonSchema != undefined) {
```
Comment (trevorling) [inline GUI/src/pages/NewServicePage.tsx:22]: Isn't this enough?
```Ts
if (jsonSchema) {....}

File: GUI/src/pages/OverviewPage.tsx
Comment (trevorling) [inline GUI/src/pages/OverviewPage.tsx:1]: Please revert back to single quotes.

File: GUI/src/types/endpointType.ts
Comment (trevorling) [inline GUI/src/types/endpointType.ts:1]: We have `routes-constants.ts` and here the file name is `endpointType.ts` - which naming convention we are using?

**Review by trevorling — CHANGES_REQUESTED** [review]

File: GUI/src/resources/api-constants.ts
Comment (baha-a) [inline GUI/src/resources/api-constants.ts:1]: Please add newline at end of file.

File: DSL/Ruuter/POST/services/open-api-spec.yml
Comment (baha-a) [inline DSL/Ruuter/POST/services/open-api-spec.yml:1]: Please add newline at end of file.

**Review by baha-a — APPROVED** [review]

**Review by trevorling — APPROVED** [review]

**Pull request merged**